orchagent/aclorch.cpp

```diff
@@ -2023,6 +2023,18 @@ bool AclRuleMirror::validate()
     return true;
 }
 
+bool AclRuleMirror::createCounter()
+{
+    SWSS_LOG_ENTER();
+
+    // If the mirror session is inactive, do not create the ACL counter from the first place
+    bool state = false;
+
+    m_pMirrorOrch->getSessionStatus(m_sessionName, state);
+
+    return !state ? true : AclRule::createCounter();
```

Comment:
This logic is a little confusing. Can you please write it in a better way?

Reply:
If the `createCounter` function is overridden, do you also override the `removecounter` function for AclRuleMirror?

Reply:
Thank you for the review. Code refactoring has been done (kindly see new commit).
`AclRuleMirror::createCounter` leverages its functionality to` AclRule::createCounter` iff the mirror session is active; otherwise, it performs NOP.
I haven't override `removeCounter` since it would behave exactly the same as `AclRule::removeCounter`.